File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>



int main(void)
{
    float sen = 0, ch = 1, n = 0, wor = 1;
    string txt = get_string("Text: \n");
    n = strlen(txt); // to get the lenght of whole text.
    for (int i = 0; i < n ; i++)
    {

        if (txt[i] == ' ') //For counting words by looking for spaces
        {
            wor++;
        }


        if (txt[i] == '!' || txt[i] == '?' || txt[i] == '.') //for counting sentences
        {

                sen++;
        }

    }

    ch = n - (wor + sen); // to get no. of characters
    float L = 0, S = 0;
    L = (ch / wor) * 100;
    S = (sen / wor) * 100;
    float index = (0.0588 * L) - (0.296 * S) - 15.8; // formula to get grade

    if (index >= 8 && index < 9)
    {
        printf("Grade 8\n");
    }
    else if (index >= 1 && index < 16)
    {
        printf("Grade %.0f\n", index);
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }

}

File: pset6/readability.py
from cs50 import get_string, get_int, get_float
import math

sentences, words = 0.0, 1.0

text = get_string("Text: ")

for word in text:  # to iterate over each word in text
    for char in word:  # to iterate over each char of the word and check for the desired details
        if char in [".", "!", "?"]:
            sentences += 1
        if char in [" "]:
            words += 1

n = len(text)  # to get the length of text
ch = n - (words + sentences)  # to get no. of characters
L = (ch / words) * 100  # L is the average number of letters per 100 words in the text
S = (sentences / words) * 100  # S is the average number of sentences per 100 words in the text
index = (0.0588 * L) - (0.296 * S) - 15.8  # formula to get grade

if (index >= 4 and index < 5):  # to print out grades in accordance with the result index.
    print("Grade 5")
elif (index >= 1 and index < 16):
    print(f"Grade {int(index)}")
elif (index < 1):
    print("Before Grade 1")
else:
    print("Grade 16+")